src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+
+/**
+ * HttpExceptionListener.
+ *
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    protected $container;
+    protected $debug;
+
+    public function __construct(ContainerInterface $container, $debug = false)
+    {
+        $this->container = $container;
+        $this->debug = $debug;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+          KernelEvents::EXCEPTION => array(
+            array('onKernelException', -256),
+          ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if (!$this->debug) {
```

Comment:
I'd suggest an alternative logic in the method:
when the event already contains a response, this listener should just do nothing.
otherwise and if possible, it would be great if creating the response could be delegated to then `ExceptionHandler` (the Debug component is already a required dependency)